index.js

```diff
@@ -794,628 +804,760 @@ function commonObject (res, id, msgId, status, resCode, err, errmsg, data) {
-        // making api call to create category
-        const cdata = await getResponseData(req, categoryUrl, createRelatedDiscussions, body, constants.post);
-        
-        if(cdata && cdata.payload) {
-          console.log('category created successfully and category id is', cdata.payload.cid)
-          const context = payload.context;
-          if(context && context.length > 0) {
-            let forumIds = [];
-            for(let i=0; i < context.length; i++) {
-              // mapping cid with context
-              const isCidMapped = await sbCategoryModel.find({sbIdentifier: context[i].identifier, sbType: context[i].type})
-              let mapObj = {
-                    sbIdentifier: context[i].identifier,
-                    sbType: context[i].type,
-                    cid: cdata.payload.cid
-                }
-              const SbObj = new sbCategoryModel(mapObj);
-              const mapResponse = await SbObj.save();
-              // getting all mapped cids for a context
-              const mappedCids = await sbCategoryModel.find({sbIdentifier: context[i].identifier, sbType: context[i].type})
-              const listOfCids = mappedCids.map(forum => forum.cid);
-              const mapResObj = {
-                "sbType": context[i].type,
-                "sbIdentifier": context[i].identifier,
-                "newCid": cdata.payload.cid,
-                "cids": listOfCids
-              }
-              forumIds.push(mapResObj);
-              if(i === (context.length - 1)){
-                const result = {forums: forumIds};
-                console.log(result)
-                res.send(responseData(req,res,createRelatedDiscussions,result, null)); 
-              }
-            }
-            // copy privileges from a category 
-           if(payload.privileges.copyFromCategory){
-              const result = await Categories.copyPrivilegesFrom(payload.privileges.copyFromCategory, cdata.payload.cid);
+  const payload = { ...req.body.category }
+  if (payload) {
+    console.log('Creating new category')
+    // creating payload for category
+    const body = {
+      parentCid: payload.pid,
+      name: payload.name || constants.defaultCategory
+    }
+    // category request url
+    const categoryUrl = `${constants.createCategory}?_uid=${payload.uid}`
+    // making api call to create category
+    const cdata = await getResponseData(
+      req,
+      categoryUrl,
+      createRelatedDiscussions,
+      body,
+      constants.post
+    )
+
+    if (cdata && cdata.payload) {
+      console.log(
+        'category created successfully and category id is',
+        cdata.payload.cid
+      )
+      const context = payload.context
+      if (context && context.length > 0) {
+        let forumIds = []
+        for (let i = 0; i < context.length; i++) {
+          // mapping cid with context
+          const isCidMapped = await sbCategoryModel.find({
+            sbIdentifier: context[i].identifier,
```

Comment:
sbIdentifier: context[i].identifier,
            sbType: context[i].type
            
So many times this has declared. Define only once with object holding this data

For all the following use cases the the object or extend the object

Reply:
This code was already there. I have not added this.

Reply:
When you notice the code, which is bad. You improve the code irrespective of who wrote the code. Then only we can improve the coding standards.
@VenkatKadiveti @bandana-sahu @christyfernandes